tests/callbacks/test_EnqueuedEventTimer.cpp

```diff
@@ -22,44 +23,90 @@ namespace Kokkos::utils::tests::callbacks
 
 using namespace Kokkos::utils::callbacks;
 
-struct EnqueuedEventTimerTest : public ::testing::Test
+template <typename TimerType>
+struct TimerTest : public ::testing::Test
 {
-public:
-    using timer_t = EnqueuedEventTimer<execution_space>;
-
 public:
     void SetUp() override {
         this->exec = Kokkos::Experimental::partition_space(execution_space{}, 1)[0];
         this->dev_id = Kokkos::Tools::Experimental::device_id(this->exec);
-        this->timer = timer_t{this->exec};
+        this->timer = TimerType{this->exec};
     }
 
 protected:
     execution_space exec {};
     uint32_t dev_id = 0;
-    timer_t timer {};
+    TimerType timer {};
 };
 
-//! @test Check @ref Kokkos::utils::callbacks::EnqueuedEventTimer.
+struct EnqueuedEventTimerTest : public TimerTest<EnqueuedEventTimer<execution_space>> {};
+
+//! @test Check the duration reported by @ref Kokkos::utils::callbacks::EnqueuedEventTimer.
 TEST_F(EnqueuedEventTimerTest, duration)
 {
     this->timer.start(BeginParallelForEvent{.name = "my-name", .dev_id = this->dev_id, .event_id = 1});
 
-    std::this_thread::sleep_for(std::chrono::milliseconds(100));
+    //std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+    this->timer.stop(EndParallelForEvent{.event_id = 1});
+
+    const double elapsed = this->timer.template duration<Kokkos::utils::timer::milliseconds>().count();
+
+    //! Check that the elapsed time is greater than 0.
+    ASSERT_GE(elapsed, 0.);
+}
+
+struct EnqueuedEventWithLaunchTimerTest : public TimerTest<EnqueuedEventWithLaunchTimer<execution_space>> {};
+
+//! @test Check the duration and launch reported by @ref Kokkos::utils::callbacks::EnqueuedEventWithLaunchTimer.
+TEST_F(EnqueuedEventWithLaunchTimerTest, duration)
+{
+    using unit_t = Kokkos::utils::timer::microseconds;
+    
+    Kokkos::utils::timer::Timer<void>            timer_external_h {};
+    Kokkos::utils::timer::Timer<execution_space> timer_external_d {};
```

Comment:
```suggestion
    Kokkos::utils::timer::Timer<void>            timer_outer_h {};
    Kokkos::utils::timer::Timer<execution_space> timer_inner_d {};
```

* outer because it'w wrapping the whole thing
* inner because it's within the timer we're testing